edge/stack/stack.go

```diff
@@ -151,14 +152,20 @@ func (manager *StackManager) processStack(stackID int, version int) error {
 	folder := fmt.Sprintf("%s/%d", agent.EdgeStackFilesPath, stackID)
 	fileName := "docker-compose.yml"
 	fileContent := stackConfig.FileContent
-	if manager.engineType == EngineTypeKubernetes {
+
+	if manager.engineType == EngineTypeDockerStandalone {
+		if len(stackConfig.RegistryCredentials) > 0 && strings.HasPrefix(stackConfig.Name, "edge-update-schedule") {
+			yml := yaml.NewYAML(fileContent, stackConfig.RegistryCredentials)
+
+			fileContent, _ = yml.AddCredentialsAsEnvForSpecificService("updater")
+		}
+	} else if manager.engineType == EngineTypeKubernetes {
 		fileName = fmt.Sprintf("%s.yml", stack.Name)
 		if len(stackConfig.RegistryCredentials) > 0 {
 			yml := yaml.NewYAML(fileContent, stackConfig.RegistryCredentials)
 			fileContent, _ = yml.AddImagePullSecrets()
 		}
-	}
-	if manager.engineType == EngineTypeNomad {
+	} else if manager.engineType == EngineTypeNomad {
```

Comment:
why need `else`? engine type can be both

Reply:
You meant engine type can be `EngineTypeKubernetes` and `EngineTypeNomad` at the same time?

Reply:
no, I meant that this change is not needed. else is not needed here

```
if a=0 {
 ... do something
} else if a=1 {
	... do something else {
} else if a=2 {...}
```

if like
```
if a=0 {
 ... do something
} 
if a=1 {
	... do something else {
} 
if a=2 {...}
```

a switch case is probably the best use

Reply:
Three `if` will check the condition three times no matter whether either of them matches or not. `if ... else if ..` will skip the rest `if` once the condition is true. In this case, `switch case break` has the same effect.